source/jormungandr/jormungandr/realtime_schedule/timeo.py

```diff
@@ -181,21 +184,45 @@ def _get_next_passage_for_route_point(
         if not r:
             return None
 
-        if r.status_code != 200:
-            # TODO better error handling, the response might be in 200 but in error
-            logging.getLogger(__name__).error(
-                'Timeo RT service unavailable, impossible to query : {}'.format(r.url),
-                extra={'rt_system_id': unicode(self.rt_system_id), 'status_code': r.status_code},
-            )
-            raise RealtimeProxyError('non 200 response')
+        return self._get_passages(r, current_dt, route_point.fetch_line_uri())
+
+    def _get_passages(self, response, current_dt, line_uri=None):
 
-        return self._get_passages(r.json(), current_dt, route_point.fetch_line_uri())
+        status_code = response.status_code
+        timeo_resp = response.json()
 
-    def _get_passages(self, timeo_resp, current_dt, line_uri=None):
         logging.getLogger(__name__).debug(
             'timeo response: {}'.format(timeo_resp), extra={'rt_system_id': unicode(self.rt_system_id)}
         )
 
+        # Handling http error
+        if status_code != 200:
+            logging.getLogger(__name__).error(
+                'Timeo RT service unavailable, impossible to query : {}'.format(response.url),
+                extra={'rt_system_id': unicode(self.rt_system_id), 'status_code': status_code},
+            )
+            raise RealtimeProxyError('non 200 response')
+
+        # internal timeo error handling
+        message_responses = timeo_resp.get('MessageResponse')
+        for message_response in message_responses:
+            if (
+                'ResponseCode' in message_response
+                and message_response['ResponseCode'] >= self.INTERNAL_TIMEO_ERROR_CODE_LIMIT
+            ):
+                resp_code = message_response['ResponseCode']
+                if 'ResponseComment' in message_response:
+                    resp_comment = message_response['ResponseComment']
+                else:
+                    resp_comment = ''
+                logging.getLogger(__name__).error(
+                    'Timeo RT internal service error, code: {} - comment: {}'.format(resp_code, resp_comment)
```

Comment:
minor to avoid confusion between HTTP code and ResponseCode:
```suggestion
                    'Timeo RT internal service error, ResponseCode: {} - ResponseComment: {}'.format(resp_code, resp_comment)
```